### DISCUSSION TITLE
Is there a way to get information about Metaspace memory?

### DISCUSSION THREAD

**Discussion by LeonardoIPM** [javamelody/javamelody#1216] [Q&A] [answered]
I need to get metrics about the metaspace usage memory. 
In JConsole it seems to be possible, can i get this value by JavaMelody???

**Comment by evernat** [comment, marked as answer]
There is no graph in javamelody about Metaspace, but javamelody can display MBeans.
For example, see the following page in the demo and open java.lang and then Metaspace:
http://javamelody.org/demo/monitoring?part=mbeans

You will see:
```
Metaspace
type=MemoryPool
-  CollectionUsage	null	 
-  CollectionUsageThresholdSupported	false	 
-  MemoryManagerNames	[Metaspace Manager]	 
-  Name	Metaspace	 
-  ObjectName	java.lang:type=MemoryPool,name=Metaspace	 
-  PeakUsage	{committed=72 781 824, init=0, max=-1, used=70 921 824}	 
-  Type	NON_HEAP	 
-  Usage	{committed=72 781 824, init=0, max=-1, used=70 847 368}	 
-  UsageThreshold	0	 
-  UsageThresholdCount	0	 
-  UsageThresholdExceeded	false	 
-  UsageThresholdSupported	true	 
-  Valid	true
```

And http://javamelody.org/demo/monitoring?jmxValue=java.lang:name=Metaspace,type=MemoryPool.Usage returns:
`{committed=72781824, init=0, max=-1, used=71141624}`
